### ISSUE TITLE
[feature] multiple tags filtering

### ISSUE THREAD

**Issue by erixtekila** [wallabag/wallabag#1197]
While the archived content will become quiet large, finding appropriate content would become hard.
Tags are a nice way to organize content, but there are no way to search for content matching multiple tags at once.

At last, it would be cool to be able to save custom search, creating groups of content, in prder to retrieve it later.

**Comment by tcitworld** [comment]
With v2, we can provide a full features search.

**Comment by brabalan** [comment]
It would be great if this multiple tags filtering could be integrated with the existing filters, in particular to restrict to unread items.

**Comment by anarcat** [comment]
v2 is out, yet this is not resolved yet. with large library, this is a critically important feature to find specific content. for example, i manage a large collection of links to different programs under the `software` tag, but then i get too much data there - i need to filter again on another tag, something that wallabag doesn't currently allow.

i just need to have the possibility of doing the intersection of multiple tags - is this something that is considered?

thanks!

**Comment by 5TDg9** [comment]
What happend in this issue... This is a must have feature.
At the moment tags shouldn't be called "tags", but folders or something else.

Please add Support for filtering or searching mutiple tags.
You could also add operators like AND, OR or NOT for this.

**Comment by fleskesvor** [comment]
I agree that this feature would be very useful. Ideally, it should be possible to match search strings to both tags and content, but being able to search for multiple tags would already be a major improvement. As it is, I find it very difficult to find anything without going through multiple pages of search results.

**Comment by nicosomb** [comment]
> What happend in this issue... This is a must have feature.

It happens that we are a volunteer team. Have a nice day.

**Comment by 5TDg9** [comment]
I meant issue more in the sense of "in this case", of course.

Didn't want to bother you, nicosomb. It's not my duty to comment anything here at Github. Actually, I'm also a volunteer, even though I have not made a big contribution so far.

Maybe you could tell us how we can help or where to start of ?

**Comment by meonkeys** [comment]
New idea: #6100 . Feedback welcome!

**Comment by bcutter** [comment]
So how to filter for more than one tag in 2025? Guys it's been 10 years. Came here because of a search engine result pointing at https://www.reddit.com/r/selfhosted/comments/1auszxd/migrate_from_wallabag_to_what/.

**Comment by PackElend** [comment]
Same question here.
My collection got too huge to find things again by a single tag.

That is reason to quite wallabag.it and search for alternatives.

**Comment by Kdecherf** [comment]
Hello everyone,
cc @PackElend, @bcutter 

I've worked on a private patch years ago to include this feature on my own instance, kind of.
I'm now considering pushing this as a proposal for resolving this request.

Here's how my patch is currently working.

It adds a new "tag" filter in the filter modal:
<img width="254" height="186" alt="Image" src="https://github.com/user-attachments/assets/8afca565-9583-41e2-a669-a4d98080e07b" />

The available operators are:

* Match a given tag (_exact match_), e.g. `software` (_referred to inclusion condition below_)
* Not match a given tag by prepending the minus sign to the tag name (_exact match_), e.g. `-software` (_referred to exclusion condition below_)

The patch additionally supports limited pattern matching using `%`, I'm using this mostly for tag grouping.
Consider that I have `topic:software` and `topic:hardware`, I can use `topic:%` to match both. It also works with the exclusion condition, e.g. `-topic:%` would exclude all entries matching this pattern.

You can add several conditions by using a comma, for example: `software,-hardware`.

The final expression can be summarized as: "(match ANY inclusion conditon) AND (do not match ALL exclusion conditions)"
This means that the expression `software,hardware` would return all articles having either the `software` or the `hardware` tag.

I may change "match ANY inclusion condition" to "match ALL inclusion conditions" if requested, however I do not plan to work on a more complex approach with AND and OR operators for this first version.

Do you have any feedback on this approach and proposal?

**Comment by PackElend** [comment]
> I've worked on a private patch years ago to include this feature on my own instance, kind of.
> I'm now considering pushing this as a proposal for resolving this request.

that is great news 🎉 


> Do you have any feedback on this approach and proposal?

The challenge I face is that, at a certain point, the collection becomes too large for me to remember all my tags.
To mitigate this, I ensure that each entry has a generic tag. Other apps, such as Linkkwarden or Karakeep, use 'Collection' or 'Lists' for this purpose.
Ideally, I would like to be able to click on a tag on the 'tag list' page at https://app.wallabag.it/tag/list, or type it into the search bar.  
On the resulting list, I would like to be able to select a second tag by pressing Ctrl and clicking the mouse, for example, to show all entries containing these two tags. 
I would then select a third tag, and so on.Ideally, I would also like to be able to select multiple tags in https://app.wallabag.it/tag/list.

If that is not possible, an alternative would be for the above click operation to add the selected tags to the special search window.

I try to explain it by an example. 
I'm looking for an article about a restaurant that offers great organic and healthy food. On their website, they explain how things are created (so you can cook them yourself, in theory).

How can I find that article? I only remember it being about restaurants and health.

that could be listed in...
Travel?
nutrition?
Health?
(Restaurants is rather a subcategory of travel).

If I search under 'Travel', I might find it, but there could be a relevant article with the same tags as 'Healthy' and 'Cook', so I select 'Health' and 'Healthy' and 'Cook' to find any related entries.



The point is to establish an efficient way to browse tags without having to know each one by heart. It requires a certain degree of discipline to tag things in the first place, of course.
